packages/use-growthbook/src/AbTestProvider.tsx

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-expect-error TODO: remove once Growthbook is correctly typed and export
+import { GrowthBook, GrowthBookProvider } from '@growthbook/growthbook-react'
+import type { ReactNode } from 'react'
+import { useCallback, useEffect, useMemo } from 'react'
+import type { GrowthBookType } from './types'
+
+export type ToolConfig = {
+  apiHost: string
+  clientKey: string
+  enableDevMode: boolean
+}
+
+export type TrackingCallback = (
+  experiment: { key: string },
+  result: { key: string },
+) => null
+
+export type AbTestProviderProps = {
+  children: ReactNode
+  anonymousId: string
+  config: ToolConfig
+  trackingCallback: TrackingCallback
+  errorCallback: (error: string) => null
+}
+
+const getGrowthBookInstance = ({
+  config: { apiHost, clientKey, enableDevMode },
+  anonymousId,
+  trackingCallback,
+}: {
+  config: ToolConfig
+  anonymousId: string
+  trackingCallback: TrackingCallback
+}): GrowthBookType =>
+  new GrowthBook({
+    apiHost,
+    clientKey,
+    enableDevMode,
+    attributes: {
+      anonymousId,
+      userId: undefined,
+      organizationId: undefined,
+      organizationType: undefined,
+    },
+    trackingCallback,
+  })
+export function AbTestProvider({
```

Comment:
I saw both on the monorepo so I didn't know what to choose. Any recommendation ?

Reply:
IIRC function were used because typescript didn't support generic const function declaration, that has been fixed on their side a while ago, I'll try to convert them to const :)

But yeah, we favour const anywhere we can